### PR TITLE
Upadte TESTING.ascidoc BWC section

### PR DESCRIPTION
**Pull request by alpar-t**
- Remove outdated sections
-  add pointer to VersionCollections from #34818.
- align some headers

### CODE REVIEW

File: TESTING.asciidoc [hunk 1]
```diff
@@ -554,25 +526,26 @@ environment variable.
...
 -------------------------------------------------
 ./gradlew v5.3.2#bwcTest
 -------------------------------------------------
 
-When running `./gradlew check`, some minimal backcompat checks are run. Which version
-is tested depends on the branch. On master, this will test against the current
-stable branch. On the stable branch, it will test against the latest release
-branch. Finally, on a release branch, it will test against the most recent release.
+Tests are ran for versions that are not yet released but with which the current version will be compatible with.
+These are automatically checked out and build from source.
```
Comment (rjernst) [inline TESTING.asciidoc]: build -> built

Comment (rjernst) [reply TESTING.asciidoc]: I think it would be useful to point to the distribution/bwc project that roots all the bwc build checkouts.

**Review by rjernst — APPROVED** [review]
LGTM, left a suggestion

**Review by nik9000 — APPROVED** [review]
I'm happy with it once @rjernst likes it.

**Comment by nik9000** [comment]
And he approved it too. So LGTM.

**Pull request merged**